multi-educator/src/Dscp/MultiEducator/Launcher/Resource.hs

```diff
@@ -62,6 +65,26 @@ deriveHasLensDirect ''MultiEducatorResources
 deriveHasLens 'merWitnessResources ''MultiEducatorResources ''Witness.WitnessResources
 deriveHasLens 'merWitnessResources ''MultiEducatorResources ''NetServResources
 
+instance AllocResource EducatorContextsVar where
+    type Deps EducatorContextsVar = ()
+    allocResource () =
+        fmap EducatorContextsVar $
+        buildComponentR "Educator contexts var"
+            (newTVarIO mempty)
+            shutdownAllContexts
+      where
+        shutdownAllContexts ctxsVar = do
+            ctxs <- atomically $ readTVar ctxsVar
+            -- TODO [DSCP-494]: prevent further insertions to the context var
```

Comment:
I wonder if doing this in one transaction would suffice.

Reply:
What do you mean?

Reply:
I mean, reading TVar and shutting down contexts in one STM transaction.

Reply:
Ah right, I'm already doing so in DSCP-494 (yet unpushed), so it will be alright eventually.